server/src/main/scala/com/cloudera/livy/server/batch/BatchSession.scala

```diff
@@ -62,7 +63,11 @@ object BatchSession {
           request.conf, request.jars, request.files, request.archives, request.pyFiles, livyConf))
       require(request.file != null, "File is required.")
 
-      val builder = new SparkProcessBuilder(livyConf)
+      val builder = new SparkProcessBuilder(livyConf, request.sparkVersion)
+      request.sparkVersion.map({ value =>
```

Comment:
I think we will need map in builder.env as sparkConfDir is optional variable and builder.env declaration needs a string.  Agreed we can remove the first map of  request.sparkVersion. Will make the changes and deploy.